File: ClientApp/notes-app/src/app/service/post.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable, of} from "rxjs";
import {Post} from "../model/post";
import {Valutazione} from "../model/valutazione";

@Injectable({
  providedIn: 'root'
})
export class PostService {

  private postURL = 'http://localhost:8080/post'

  constructor(private http : HttpClient) { }

  //****************************************************************
  //SERVIZI HTTP POST

  getAllPosts(): Observable<Post[]>{
    return this.http.get<Post[]>(this.postURL + '/getAllPosts')
  }

  //TODO da testare
  getPostsByString(parola : string): Observable<Post[]>{
    return this.http.get<Post[]>(this.postURL + '/getPostsString/' + parola)
  }

  getPostsbyValutazione(valutazione : string): Observable<Post[]>{
    return this.http.get<Post[]>(this.postURL + '/getPostsValutazione/' + valutazione)
  }

  getPostsbyUtente(nomeUtente: string): Observable<Post[]>{
    return this.http.get<Post[]>(this.postURL + '/getPostsUtente/' + nomeUtente)
  }

  getPostByCategoria(categoria : string): Observable<Post[]>{
    return this.http.get<Post[]>(this.postURL + '/getPostsCategoria/' + categoria)
  }


  setPost(post: Post): Observable<unknown>{
    return this.http.post<Post>(this.postURL, post)
  }

  updatePostValutazione(data : Valutazione) : Observable<unknown>{
    return this.http.put(this.postURL + '/updatePostValutazione/' + data.idPost + '/' + data.valutazione.toString() + '/' + data.nomeUtente, data);
  }


  //TODO da testare
  deletePost(id: string): Observable<unknown>{
    return this.http.delete(this.postURL + '/deletePost/' + id)
  }
}

File: spring-boot-socialnotes/src/main/java/com/socialnotes/service/PostService.java
package com.socialnotes.service;

import com.socialnotes.dao.PostDao;
import com.socialnotes.model.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class PostService {
    private final PostDao postDao;
    @Autowired
    public PostService(@Qualifier("PostDao") PostDao postDao) { this.postDao = postDao; }

    public List<Post> getAllPost () { return postDao.getAllPost(); }
    public List<Post> getPostsString(String string) { return postDao.getPostsString(string); }
    public List<Post> getPostsValutazione (String valutazione) { return postDao.getPostsValutazione(valutazione); }
    public List<Post> getPostsCategoria (String categoria) { return postDao.getPostsCategoria(categoria); }
    public List<Post> getPostsUtente (String nomeUtente) { return postDao.getPostsUtente(nomeUtente); }
    public boolean setPost(Post post) { return postDao.setPost(post); }
    public boolean updatePostValutazione(String idPost, int valutazione, String nomeUtente) {
        return postDao.updatePostValutazione(idPost, valutazione, nomeUtente);
    }
    public boolean updatePostSegnalazione (String idPost){ return postDao.updatePostSegnalazione(idPost); }
    public boolean deletePost (String idPost) { return postDao.deletePost(idPost); }

}
